core/src/main/scala/zio/config/PropertyTree.scala

```diff
@@ -1,12 +1,98 @@
 package zio.config
 
-import zio.config.PropertyTree.{ Empty, Leaf, Record }
-import zio.config.PropertyTree.Sequence
+import scala.collection.immutable.Nil
+import PropertyTree._
 
 sealed trait PropertyTree[+K, +V] { self =>
+  final def map[V2](f: V => V2): PropertyTree[K, V2] = self match {
+    case Leaf(value)     => Leaf(f(value))
+    case Record(v)       => Record(v.map { case (k, tree) => (k, tree.map(f)) })
+    case Sequence(value) => Sequence(value.map(_.map(f)))
+    case Empty           => Empty
+  }
+
+  final def getOrElse[K1 >: K, V1 >: V](tree: => PropertyTree[K1, V1]): PropertyTree[K1, V1] =
+    if (self == PropertyTree.empty) tree else self
+
+  def mapEmptyToError[E, V2](f: => E): PropertyTree[K, Either[E, V]] = self match {
+    case Leaf(value)        => Leaf(Right(value))
+    case Record(v)          => Record(v.map { case (k, tree) => (k, tree.mapEmptyToError(f)) })
+    case PropertyTree.Empty => PropertyTree.Leaf(Left(f))
+    case Sequence(value)    => Sequence(value.map(_.mapEmptyToError(f)))
+  }
+
+  final def zipWith[K1 >: K, V2, V3](that: PropertyTree[K1, V2])(f: (V, V2) => V3): PropertyTree[K, V3] =
```

Comment:
We should test the following laws:

```scala
tree.zipWith(tree)(_._1) == tree
tree.zipWith(tree)(_._2) == tree
```

for all `tree`

Reply:
Sure

Reply:
Done